### PR TITLE
[BPK-1399] Add iconOnly prop to Android BpkButton

### PR DESCRIPTION
**Pull request by shaundon**

### CODE REVIEW

File: native/packages/react-native-bpk-component-button/src/BpkButton-test.common.js [hunk 1]
```diff
@@ -59,6 +59,21 @@ const commonTests = () => {
...
+      const tree = renderer
+        .create(
+          <BpkButton
+            iconOnly
+            icon="baggage"
+            title="Lorem ipsum"
+            onPress={onPressFn}
+          />,
+        )
+        .toJSON();
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-button/src/BpkButton-test.common.js]: Moved from iOS tests to common.

File: native/packages/react-native-bpk-component-button/src/BpkButton-test.ios.js [hunk 2]
```diff
@@ -54,23 +54,6 @@ describe('iOS', () => {
...
-          icon="baggage"
-          title="Lorem ipsum"
-          onPress={onPressFn}
-        />,
-      )
-      .toJSON();
-
-    expect(tree).toMatchSnapshot();
-  });
-
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-button/src/BpkButton-test.ios.js]: Test was moved to common.

File: native/packages/react-native-bpk-component-button/src/BpkButton.ios.js [hunk 3]
```diff
@@ -128,13 +127,11 @@ const BpkButton = (props: Props) => {
 
 BpkButton.propTypes = {
   ...commonPropTypes,
-  iconOnly: PropTypes.bool,
   large: PropTypes.bool,
 };
 
 BpkButton.defaultProps = {
   ...commonDefaultProps,
-  iconOnly: false,
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-button/src/BpkButton.ios.js]: Moved to common types / props.

File: native/packages/react-native-bpk-component-button/src/styles.js [hunk 4]
```diff
@@ -127,9 +127,14 @@ const modifiers = {
...
-      width: spacingXl,
-    },
+    container: Platform.select({
+      ios: () => ({
+        width: spacingXl,
+      }),
+      android: () => ({
+        width: buttonHeight,
+      }),
+    })(),
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-button/src/styles.js]: The button wasn't wide enough on Android, and therefore wasn't a circle. This makes sure it matches the button height to be round.

Comment (matthewdavidson) [reply native/packages/react-native-bpk-component-button/src/styles.js]: Begs the question, why not `buttonHeight` on both platforms?

Comment (shaundon) [reply native/packages/react-native-bpk-component-button/src/styles.js]: Tbh I left it as is, as in my head this is the button component:



But I'll give it a go now and see!

Comment (shaundon) [reply native/packages/react-native-bpk-component-button/src/styles.js]: It works, take your upvote and get out of here 🤣 (I'll push an update now!)

**Comment by shaundon** [comment]
Fixed conflicts and rebased.

**Pull request merged**